examples/gui_qt_asyncio.py

```diff
@@ -0,0 +1,74 @@
+"""
+An example demonstrating a qt app with a wgpu viz inside.
+
+This is the same as the ``gui_qt_embed.py`` example, except this uses
+the asyncio compatible mode that was introduced in Pyside 6.6.
+
+For more info see:
+
+* https://doc.qt.io/qtforpython-6/PySide6/QtAsyncio/index.html
+* https://www.qt.io/blog/introducing-qtasyncio-in-technical-preview
+
+"""
+
+# ruff: noqa: N802
+# run_example = false
+
+import time
+import asyncio
+
+from PySide6 import QtWidgets, QtAsyncio
+from wgpu.gui.qt import WgpuWidget
+from triangle import setup_drawing_sync
+
+
+class ExampleWidget(QtWidgets.QWidget):
+    def __init__(self):
+        super().__init__()
+        self.resize(640, 480)
+        self.setWindowTitle("wgpu triangle embedded in a qt app")
+
+        splitter = QtWidgets.QSplitter()
+
+        self.button = QtWidgets.QPushButton("Hello world", self)
+        self.canvas = WgpuWidget(splitter)
+        self.output = QtWidgets.QTextEdit(splitter)
+
+        # With QtAsyncio, the callbacks can now return a future. You'd
+        # think that you could also return a coro, but we need to wrap
+        # it into a future, making this code a bit ugly.
+        # self.button.clicked.connect(self.whenButtonClicked)  # why not :/
+        self.button.clicked.connect(
+            lambda: asyncio.ensure_future(self.whenButtonClicked())
```

Comment:
What?? This looks like a mistake.

Reply:
Nope, this is the actual code they use in the example in the [introductory blog post](https://www.qt.io/blog/introducing-qtasyncio-in-technical-preview):

<img width="859" alt="image" src="https://github.com/user-attachments/assets/610510f9-4243-4d64-996d-9a8030e06418">

Reply:
It's silly, because they probably use `asyncio.isfuture()` somewhere, and they could also check `asyncio.iscoroutine()` ...

Reply:
Ah, I think I get it. The `ensure_future()`  actually schedules the coro to run in whatever is the current loop.

Still would be way more natural to be able to pass a coro-function, and Qt scheduling it to run. But maybe that requires changes in Qt itself. The whole QtAsyncio is pure Python.

Reply:
Seems like they're missing a `.connect_async` method that handles the boilerplate. :)

Reply:
Refactored a little to make this code less ugly.